napalm/nxos_ssh/nxos_ssh.py

```diff
@@ -1319,6 +1320,255 @@ def process_mac_fields(vlan, mac, mac_type, interface):
+                if (not (search_re_dict['bgpie']['result'] in ['redist', 'local'])) \
+                        and (bgpnh != next_hop):
+                    # this is not the right route
+                    continue
+                # find remote AS nr. of this neighbor
+                bgpcmd = CMD_SHIBNV.format(vrf=vrf, neigh=bgpnh)
+                outbgpnei = self._send_command(bgpcmd)
+                matchbgpattr = \
+                    re.search(r"remote AS ("+ASN_REGEX+r")", outbgpnei)
+                if matchbgpattr:
+                    bgpras = matchbgpattr.group(1)
+                else:
+                    # next-hop is not known in this vrf, route leaked from
+                    #  other vrf or from vpnv4 table?
+                    # get remote AS nr. from as-path if it is ebgp neighbor
+                    # if locally sourced remote AS if undefined
+                    if search_re_dict['bgpie']['result'] == 'external':
+                        bgpras = search_re_dict['aspath']['result'].split(' ')[0].replace('(', '')
+                    elif search_re_dict['bgpie']['result'] == 'internal':
+                        bgpras = bgpas
+                    else:    # redist, local
+                        bgpras = ''
+                # community
+                bothcomm = []
+                extcomm = []
+                stdcomm = search_re_dict['bgpcomm']['result'].split()
+                commsplit = bgppath.split("Extcommunity:")
+                if len(commsplit) == 2:
+                    for line in commsplit[1].split('\n')[1:]:
+                        #          RT:65004:22
+                        matchcommun = re.match(r"[ ]{10}([\S ]+)", line)
+                        if matchcommun:
+                            extcomm.append(matchcommun.group(1))
+                        else:
+                            # we've reached end of the extended community section
+                            break
+                bothcomm = stdcomm + extcomm
+                bgp_attr = {
+                    "as_path": search_re_dict['aspath']['result'].strip(),
+                    "remote_address": search_re_dict['bgpfrom']['result'],
+                    "local_preference": int(search_re_dict['bgplp']['result']),
+                    "communities": bothcomm,
+                    "local_as": napalm.base.helpers.as_number(bgpas)
+                }
+                if bgpras:
+                    bgp_attr['remote_as'] = napalm.base.helpers.as_number(bgpras)
+                else:
+                    bgp_attr['remote_as'] = 0  # 0? , locally sourced
+        return bgp_attr
+
+    def get_route_to(self, destination='', protocol=''):
+        '''
+        Only IPv4 supported, vrf aware, longer_prefixes parameter ready
+        '''
+        RE_VIASTR = re.compile(r"    ([\*| ])via (("+IPV4_ADDR_REGEX+r")(%(\S+))?, )?"
+                               r"(([\w./:]+), )?\[(\d+)/(\d+)\], ([\d\w:]+), ([\w]+)(-(\d+))?(.*)")
```

Comment:
You should compile it outside of the function, it can be either a class or a module constant.